src/types/result.rs

```diff
@@ -74,7 +74,8 @@ pub struct CheckResult {
     pub status_reason: Option<CheckStatusReason>,
 
     /// Trace ID associated with the check-in made
-    pub trace_id: TraceId,
+    #[serde(serialize_with = "uuid_simple")]
+    pub trace_id: Uuid,
```

Comment:
I couldn't instantiate `TraceId` correctly since it's private, but I don't think it adds too much here since it's basically just a uuid